There isn't a member named 'it'. The correct member name is 'm_it'.
--- a/common/EntityListHelpers.cpp
+++ b/common/EntityListHelpers.cpp
@@ -98,7 +98,7 @@ struct EntityMapTypedIterator
 
     bool operator==(EntityMapTypedIterator const &other) const
     {
-        return m_it == other.it;
+        return m_it == other.m_it;
     }
 
     template <typename Y>
